scripts/services/check-removed-services.js

```diff
@@ -0,0 +1,138 @@
+    }
+
+    /**
+     * Retrieves the names of old services after normalization.
+     *
+     * @returns {Promise<Array<string>>} An array of normalized old service names.
+     */
+    const getOldServicesNames = async (oldData) => {
+        // get array with id's from old data
+        const oldServicesNameArray = oldData.map((service) => service.id);
+        // format file names
+        const formattedServiceNames = oldServicesNameArray.map((serviceName) => normalizeFileName(serviceName));
+        // sort array by name
+        formattedServiceNames.sort();
+        return formattedServiceNames;
+    };
+
+    /**
+     * Retrieves the names of new services from the specified folder after normalization.
+     *
+     * @param {string} folder - The path to the folder containing new service files.
+     * @returns {Promise<Array<string>>} An array of normalized new service names.
+     */
+    const getNewServicesNames = async (distPath) => {
+        // get all dir names from services folder
+        const NewServicesFileNames = await fs.readdir(distPath);
+        const NewServicesNameArray = NewServicesFileNames.map((file) => {
+            // get only names without extension
+            const onlyName = file.replace('.yml', '');
+            return onlyName;
+        });
+        // format file names
+        const formattedServiceNames = NewServicesNameArray.map((serviceName) => normalizeFileName(serviceName));
+        // sort array by name
+        formattedServiceNames.sort();
+        return formattedServiceNames;
+    };
+
+    const newServicesNames = await getNewServicesNames(distFolder);
+    const oldServicesNames = await getOldServicesNames(oldServicesData);
+
+    const differences = oldServicesNames.filter((item) => !newServicesNames.includes(item));
+
+    /**
+     * Rewrites YAML files for removed services.
+     *
+     * @param {Array<string>} removedServices - An array of removed service names.
+     */
+    const rewriteYMLFile = async (removedServices) => {
+        try {
+            // Get objects from the old data, that have been deleted.
+            const onlyRemovedObjects = removedServices
+                .map((removedService) => {
+                    const serviceItem = oldServicesData
+                        .find((service) => normalizeFileName(service.id) === removedService);
+                    return serviceItem;
+                });
+            /**
+             * Write removed services into YML files.
+             * @param {Array<object>} removedObjects - Array of objects that should be written in separate YML files.
+             */
+            const writeRemovedServices = async (removedObjects) => {
+                if (removedObjects.length === 0) {
+                    return;
+                }
+                const [removedObject, ...restObjects] = removedObjects;
+                await fs.writeFile(
+                    path.join(`${servicesDir}/${removedObject.id}.yml`),
```

Comment:
maybe it would be better to create a const for the extension at the beginning of the script:
```js
const EXT_YML = '.yml';
```